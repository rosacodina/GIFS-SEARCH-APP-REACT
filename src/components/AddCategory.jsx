import { useState } from "react";

const AddCategory = ( {onNewCategory}) => {

  const [ inputValue, setInputValue ] = useState( "" );

  const onInputChange = ( event ) => {
    setInputValue( event.target.value );
  }

  const onSubmit = ( event ) => {
    event.preventDefault();
    if ( inputValue.trim().length <= 1) return;

    setInputValue( " " );
    onNewCategory( inputValue.trim() );
  }

  return (
    <form onSubmit={  ( event => onSubmit( event ) )}>

      <input
        type = "text"
        placeholder= "Busca un gif"
        value = { inputValue }
        onChange = { (event) => onInputChange( event ) }
      />

    </form>
    
  )
}

export {AddCategory}
