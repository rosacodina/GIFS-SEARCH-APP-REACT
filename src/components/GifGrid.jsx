import {GifItem} from "./GifItem";
import { useFetchGifs } from "../components/hooks/useFetchGifs";

const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs( category );
  
  return (
    <>
      <h3> { category } </h3>
      {
        isLoading && ( <h3>Cargando...</h3> )
      }
      

      <div className="card-grid">
        {
          images.map( (image) => (
            <GifItem 
              key={ image.id }
              { ...image }
            />
          ))
        }
      </div>
    </>
  )
}

export {GifGrid};
