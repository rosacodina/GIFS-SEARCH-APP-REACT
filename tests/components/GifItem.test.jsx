/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en GifItem component', () => {

  const title = "Miércoles";
  const url = "https://one-punch.com/miercoles.jpg"

  test('debe coincidir con el snapshot ', () => {

    const { container } = render(<GifItem title={ title } url={ url } />);
    expect( container ).toMatchSnapshot();

  });

  test('debe enseñar la imagen con el URL y el ALT indicado ', () => {
    
    render(<GifItem title={ title } url={ url } />);
    //screen.debug();
    const { src, alt } = screen.getByRole("presentation");
    expect( src ).toBe( url );
    expect( alt ).toBe( alt );
  });

  test('debe enseñar el título en el componente ', () => {
    
    render(<GifItem title={ title } url={ url } />);
    expect( screen.getByText( title ) ).toBeTruthy();

  });
  
     
});
