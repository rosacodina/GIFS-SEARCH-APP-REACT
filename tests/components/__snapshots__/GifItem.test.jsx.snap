// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem component debe coincidir con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Miércoles"
      src="https://one-punch.com/miercoles.jpg"
    />
    <p>
      Miércoles
    </p>
  </div>
</div>
`;
